lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -33,13 +38,26 @@ def buyable_trains(entity)
             trains_to_buy = super
             trains_to_buy = trains_to_buy.select(&:from_depot?) unless @game.can_buy_trains
             trains_to_buy = trains_to_buy.reject { |t| t.name == '2E' } if owns_2e?(entity)
-            trains_to_buy
+            fix_2e_issue(entity, trains_to_buy)
           end
 
           def owns_2e?(entity)
             entity.trains.any? { |t| t.name == '2E' }
           end
 
+          def fix_2e_issue(entity, trains_to_buy)
```

Comment:
Discounts are handled implicitly. Why do you need this?

Reply:
the Ghan costs $200, the privates give $100 discount, If a corp has 100-199 left in the bank the 2e doesn't show up despite having the ability to pay for it

Reply:
That capability is built-in. You shouldn't need special code for it. See American Locomotive private in 18USA as an example.

Reply:
Example of American Locomotive discounting: https://18xx.games/game/92368?action=322

Reply:
I'll try to figure out how to fix this bug in the base code